src/ol/source/tileutfgridsource.js

```diff
@@ -271,7 +271,7 @@ ol.source.TileUTFGridTile_.prototype.getImage = function(opt_context) {
  */
 ol.source.TileUTFGridTile_.prototype.getData = function(coordinate) {
   if (goog.isNull(this.grid_) || goog.isNull(this.keys_) ||
-      goog.isNull(this.data_)) {
+      !goog.isDefAndNotNull(this.data_)) {
```

Comment:
An alternative is to use `Object.<string, Object>` for the type for `this.data_` and use the following when setting `this.data_`:

```js
this.data_ = goog.isDef(json.data) ? json.data : null;
```